qa/rpc-tests/finalsaplingroot.py

```diff
@@ -206,14 +206,25 @@ def run_test(self):
 
         assert_equal(len(self.nodes[0].getblock("206")["tx"]), 2)
         assert_equal(self.nodes[0].z_getbalance(taddr2), Decimal("2.34"))
+        assert_equal(self.nodes[1].z_getbalance(saplingAddr1), 0)
 
+        # Verify the final Sapling root changes (because the Sapling bundle was padded
+        # with 2 dummy outputs).
         blk = self.nodes[0].getblock("206")
         root = blk["finalsaplingroot"]
-        assert_equal(root, self.nodes[0].getblock("205")["finalsaplingroot"])
+        assert root is not self.nodes[0].getblock("205")["finalsaplingroot"]
```

Comment:
```suggestion
        assert root != self.nodes[0].getblock("205")["finalsaplingroot"]
```

`is not` means reference inequality:
```
$ python
Python 3.11.2 (main, Mar  5 2023, 08:28:49) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> [1,2,3] is [1,2,3]
False
>>> [1,2,3] is not [1,2,3]
True
```

Reply:
Ah, hmm. That's a bug in the test as a whole then, because I copied this from other lines of the test. I'll fix it consistently in a separate commit.